### PR TITLE
Drive fixes overlayfs

### PR DESCRIPTION
**Pull request by akallabeth**
* Fixes improper ```fopen``` mode.  ```FileGetMode``` must not check the mask define ```STANDARD_RIGHTS_WRITE``` for mode conversion.
* Speed improvement, removed the intermediate buffer for ```drive_process_irp_read```
* Additional argument checks for drive channel
* Fixed type conversion warnings when encountered.

### CODE REVIEW

File: channels/drive/client/drive_main.c [hunk 1]
```diff
@@ -64,6 +64,8 @@ struct _DRIVE_DEVICE
 	rdpContext* rdpcontext;
 };
 
+static UINT sys_code_page = 0;
```
Comment (hardening) [inline channels/drive/client/drive_main.c]: too bad you've not taken that opportunity to remove that global variable

Comment (akallabeth) [inline channels/drive/client/drive_main.c]: @hardening We should do another pr for that. Remove all instances of hard coded ```CP_UTF8``` with ```GetACP()``` calls.

**Comment by akallabeth** [comment]
@freerdp-bot do me a favour and test!

**Comment by akallabeth** [comment]
@freerdp-bot please test, come on!

**Comment by akallabeth** [comment]
@hardening @bmiklautz Anyone time to review?

File: winpr/libwinpr/file/file.c [hunk 2]
```diff
@@ -669,6 +670,27 @@ static HANDLE FileCreateFileA(LPCSTR lpFileName, DWORD dwDesiredAccess, DWORD dw
...
+		 * call if not connected. This renders the channel unusable,
+		 * therefore abort early. */
+		if (S_ISFIFO(st.st_mode))
+		{
+			SetLastError(ERROR_FILE_NOT_FOUND);
+			free(pFile->lpFileName);
+			free(pFile);
+			return INVALID_HANDLE_VALUE;
+		}
+	}
```
Comment (bmiklautz) [inline winpr/libwinpr/file/file.c]: Maybe we should also not show them in file listings?

Comment (akallabeth) [inline winpr/libwinpr/file/file.c]: @bmiklautz done.

**Comment by bmiklautz** [comment]
@freerdp-bot test.

**Comment by bmiklautz** [comment]
@hardening any objections?

**Comment by akallabeth** [comment]
@freerdp-bot test please

**Comment by bmiklautz** [comment]
@freerdp-bot test.

**Comment by akallabeth** [comment]
@bmiklautz @hardening anyone time for review/merge?
